app/assets/javascripts/admin/enterprises/controllers/country_controller.js.coffee

```diff
@@ -0,0 +1,24 @@
+angular.module("admin.enterprises").controller 'countryCtrl', ($scope, availableCountries) ->
+
+  $scope.countries = availableCountries
+
+  $scope.countriesById = $scope.countries.reduce (obj, country) ->
+    obj[country.id] = country
+    obj
+  , {}
+
+  $scope.countryOnChange = (stateSelectElemId) ->
```

Comment:
I would add a comment explaining why we had to go with `setTimeout`. It'll be valuable in the future.